test/tstDetailsUtils.cpp

```diff
@@ -235,3 +236,27 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(min_and_max, DeviceType, ARBORX_DEVICE_TYPES)
   BOOST_TEST(ArborX::min(space, w) == 2);
   BOOST_TEST(ArborX::max(space, w) == 8);
 }
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(sort_objects, DeviceType, ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  ExecutionSpace space{};
+
+  for (auto values : {std::vector<int>{36, 19, 25, 17, 3, 7, 1, 2, 9},
+                      std::vector<int>{36, 19, 25, 17, 3, 9, 1, 2, 7},
+                      std::vector<int>{100, 19, 36, 17, 3, 25, 1, 2, 7},
+                      std::vector<int>{15, 5, 11, 3, 4, 8}})
+  {
+    Kokkos::View<int *, Kokkos::HostSpace> host_view("data", values.size());
+    std::copy(values.begin(), values.end(), host_view.data());
+    auto device_view = Kokkos::create_mirror_view_and_copy(space, host_view);
+    auto device_permutation = ArborX::Details::sortObjects(space, device_view);
+    Kokkos::deep_copy(space, host_view, device_view);
+    auto host_permutation = Kokkos::create_mirror_view_and_copy(
+        Kokkos::HostSpace{}, device_permutation);
+    for (unsigned int i = 1; i < values.size(); ++i)
+      BOOST_TEST(host_view(i - 1) <= host_view(i));
+    for (unsigned int i = 0; i < values.size(); ++i)
+      BOOST_TEST(values[host_permutation(i)] == host_view(i));
```

Comment:
Something like this would be better:
```suggestion
  for (auto v : {
  				 std::vector<float>{3},
  				 std::vector<float>{1., 2., 2., 2., 2., 1.},
  				 std::vector<float>{2, 36, 19, 25, 9, 3, 7, 1, 2},
  				 std::vector<float>{17., 13., 11., 9., 8., 3., 1.},
  				 std::fector<float>{1., 4., 7., 9., 10.})
  {
    auto const n = v.size();
    Kokkos::View<int *, DeviceType> values("Testing::values", n);
    Kokkos::deep_copy(space, values, Kokkos::UnmanagedView<float*, Kokkos::HostSpace>(v.data(), n);
    auto permutation = ArborX::Details::sortObjects(space, values);
    auto values_host = Kokkos::create_mirror_view_and_copy(values);
    auto permutation_host = Kokkos::create_mirror_view_and_copy(permutation);
    BOOST_TEST(std::is_sorted(values_host.data(), values_host.data() + n));
    std::vector<int> p(n);
    std::iota(p.begin(), p.end());
    std::sort(p.begin(), p.end(), [](int i, int j) { return v[i] < v[j]; });
    BOOST_TEST(permutation_host == p, tt::per_element());
```

Reply:
I have some doubts about comparing to sorts and slightly prefer just making sure that the sort is valid (also depending on if we require the sort to be stable or not, but I think shouldn't be).
If you feel strongly about it, I can change.

Reply:
One thing I dislike about the current status of the code is that if it fails, it would a) print out many failing lines, and b) require several additional steps to figure out why it failed. Ideally, you would want BOOST_TEST to print out failing array, correct array and the place where it failed. For that, you would want to used `BOOST_TEST(array1 == array2, tt::per_element);` in some form. 

For sort, you could do `BOOST_TEST(values_host == v_view, tt::per_element);` where `v_view` is the unmanaged view of `v` after calling `std::sort(v.begin(), v.end())`. For permutation, it could be similar, but we'll need to apply it to the `values_host` and compare to the original `v`.